### ISSUE TITLE
add a 404 handler for invalid routes.

### ISSUE THREAD

**Issue by reubengt** [fac19/week7-hihi#34]
currently your api returns the default express html page for endpoints that dont exist.
since this is a json api, and you would typically write frontend code to display all the errors in the same way, consider writing a middleware function for 404 errors, below all your route handlers in server.js.

the reason your errorhandler middleware doesnt get called in this case is that no routes are matched, so there is never an actual Error being thrown, or a next() being called to pass on an error to the errorhandler.

this is nicked from the express docs:
```javascript
app.use(function (req, res, next) {
  res.status(404).send({error:"Resource not found"});
});
```

or if you want to use your existing error handler instead of sending the response immediately here, you can do something like 

```javascript
server.use((req,res,next)=>{
    const error = new Error("Resource Not Found");
      error.status = 404;
      next(error);
})
```

basically this will act as your "catch all" handler... processing any requests that dont have a matching route in your api.
